### PR TITLE
MULE-9566: Content-Type set in request builder is not considered

### PR DESCRIPTION
**Pull request by canoasan**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java [hunk 1]
```diff
@@ -96,7 +96,8 @@ public HttpRequestBuilder create(MuleEvent event, String resolvedMethod, String
             }
         }
 
-        if (!event.getMessage().getOutboundPropertyNames().contains(MuleProperties.CONTENT_TYPE_PROPERTY))
+        if (!event.getMessage().getOutboundPropertyNames().contains(MuleProperties.CONTENT_TYPE_PROPERTY)
+            && !builder.getHeaders().containsKey(MuleProperties.CONTENT_TYPE_PROPERTY))
```
Comment (marianogonzalez) [inline modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java]: use static imports

File: modules/http/src/test/java/org/mule/module/http/functional/requester/HttpRequestBuilderContentTypeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return "http-request-builder-content-type-config.xml";
+    }
+
+    @Test
+    public void responseWithContentType() throws Exception
+    {
+        MuleMessage response = runFlow("requesterContentTypeClient").getMessage();
+
+        assertThat(response.getInboundPropertyNames(), hasItem(equalToIgnoringCase(MuleProperties.CONTENT_TYPE_PROPERTY)));
+        assertThat((String) response.getInboundProperty(MuleProperties.CONTENT_TYPE_PROPERTY), equalTo("text/x-json"));
```
Comment (marianogonzalez) [inline modules/http/src/test/java/org/mule/module/http/functional/requester/HttpRequestBuilderContentTypeTestCase.java]: use static imports

File: modules/http/src/test/java/org/mule/module/http/functional/requester/HttpRequestBuilderContentTypeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public DynamicPort httpListenerPort = new DynamicPort("httpListenerPort");
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "http-request-builder-content-type-config.xml";
+    }
+
+    @Test
+    public void responseWithContentType() throws Exception
```
Comment (marianogonzalez) [inline modules/http/src/test/java/org/mule/module/http/functional/requester/HttpRequestBuilderContentTypeTestCase.java]: test name is not representative. What you're really testing is that the request builder settings have precedence over the inbound properties. Provide a better test name. A brief javadoc explaining the case would also be nice ;)

File: modules/http/src/test/resources/http-request-builder-content-type-config.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <http:listener-config name="httpConf" host="localhost" port="${httpListenerPort}"/>
+
+    <http:request-config name="requesterConfig" host="localhost" port="${httpPort}"/>
+
+    <flow name="requesterContentTypeClient" >
+        <http:listener config-ref="httpConf" path="requestClient"/>
+
+        <set-payload value="{ &quot;key1&quot; : &quot;value, &quot;key2&quot; : 2 }" mimeType="application/json" encoding="UTF-8"/>
+
+        <http:request config-ref="requesterConfig" port="${httpPort}" path="requestServer" method="GET">
```
Comment (elrodro83) [inline modules/http/src/test/resources/http-request-builder-content-type-config.xml]: port is already defined in the request-config

**Pull request merged**